[EDU-3172] feat: add automate origin shield (#506)
aziontech/docs: src/content/docs/en/pages/secure-journey/advanced-configurations/automate-origin-shield.mdx

---

> title: Automate Origin Shield

Author: ```suggestion
title: How to automate Origin Shield
```

description: Create a rule to automate the use of the Origin Shield list and block potential harmful IPs.
meta_tags: edge, secure, origin shield, list, rules engine, behavior
namespace: docs_secure_automate_origin_shield
menu_namespace: secureMenu
permalink: /documentation/guides/secure/automate-origin-shield/
---

import Button from '~/components/Button.astro'

> **Origin Shield** provides a list of IP addresses used in Azion's Edge.

Reviewer: ```suggestion
**Origin Shield** provides a list of IP addresses used in Azion's Edge Network.
```

Nodes? Platform?

Reviewer 2: ou **Origin Shield** provides a list of Azion Edge Nodes' IP addresses.

<Button href="/en/documentation/products/secure/secure-infrastructure/" text="go to secure an infrastructure" variant="secondary"></Button>

> Using it benefits your application's protection, but as the list is maintained by Azion and the IP addresses in it change from time to time, you can automate a behavior using Rules Engine to guarantee you're always using the updated list.

Reviewer: ```suggestion
By activating Origin Shield, your application's is protected from a network list is maintained by Azion. The IP addresses in it change from time to time, but you can automate a behavior using Rules Engine to guarantee you're always using the updated list.
```

> Using it benefits your application's protection, but as the list is maintained by Azion and the IP addresses in it change from time to time, you can automate a behavior using Rules Engine to guarantee you're always using the updated list.

Reviewer 3: ```suggestion
By activating Origin Shield, your application is protected from a network list maintained by Azion. The IP addresses in it change from time to time, but you can automate a behavior using Rules Engine to guarantee you're always using the updated list.
```

> 
> ## Via RTM

Reviewer 3: ```suggestion

---

## Via RTM
```

To create a rule:

1. [On RTM](/en/documentation/products/guides/how-to-access-rtm/), navigate to **Edge Firewall**.
2. Select the edge firewall in which you want to configure the rule.
3. Click the **Rules Engine* tab.
4. Click the **New Rule** button.
5. Give your rule a name and, optionally, a description.
6. As a comparison operator, select **does not match**.
7. As an argument, select the `Azion Origin Shield` list.
8. In the **Behaviors** section, select **Drop (Close Without Response)** from the behavior list.
9. Click the **Save** button.

> If your application receives a request generated from an IP that isn't in the list, edge firewall will drop the request based.

Reviewer 2: ```suggestion
If your application receives a request generated from an IP that isn't in the list, the edge firewall will drop the request.
```

Reviewer 2: not sure... the original seemed weird

---

## Via API

> 1. Run the following `GET` request to retrieve the `id` of the `Azion Origin Shield` list:

Reviewer: ```suggestion
1. Run the following `GET` request to retrieve the `id` of the **Azion Origin Shield** list:
```

    ],
    "criteria": [
        [
            {
                "variable": "network",
                "operator": "is_not_in_list",
                "conditional": "if",
                "argument": "<network_list_id>"
            }
        ]
    ]
}'
```

| Key | Description |
| --- | --- |
| `name` | Name of the rule |
| `behaviors` | Array that stores objects that define behaviors |
| `criteria` | Array that stores objects that define criteria |

See the [Azion API documentation](https://api.azion.com/#ff2b221b-0054-407f-aad3-39cda3a9ed0f) to find out more about criteria and behavior objects.

4. You'll receive the following response:

```json
{
  "results": {
    "name": "Automate Origin Shield",
    "is_active": true,
    "behaviors": [
      {
        "name": "drop"
      }
    ],
    "criteria": [
      [
        {
          "variable": "network",
          "operator": "is_not_in_list",
          "conditional": "if",
          "argument": "6217"
        }
      ]
    ],
    "last_modified": "2023-11-29T19:53:56.476161Z",
    "last_editor": "user@email.com",
    "id": 28629,
    "order": 3
  },
  "schema_version": 3
}
```

5. Wait a few minutes for the changes to propagate.

> If your application receives a request generated from an IP that isn't in the list, edge firewall will drop the request based.

Reviewer: Contributors!

> If your application receives a request generated from an IP that isn't in the list, edge firewall will drop the request based.

Reviewer 3: ```suggestion
If your application receives a request generated from an IP that isn't in the list, the edge firewall will drop the request.
```

Reviewer 3: para alinhar com um comentário anterior
